pkg/monitortests/kubeapiserver/faultyloadbalancer/monitortest.go

```diff
@@ -0,0 +1,209 @@
+package faultyloadbalancer
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	MonitorName = "faulty-load-balancer"
+)
+
+var (
+	lbTypes = map[string]struct{}{
+		"internal-lb":     {},
+		"external-lb":     {},
+		"service-network": {},
+	}
+)
+
+func NewMonitorTest() monitortestframework.MonitorTest {
+	return &monitorTest{}
+}
+
+type monitorTest struct {
+	monitor *faultyLBMonitor
+}
+
+func (test *monitorTest) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	test.monitor = &faultyLBMonitor{}
+	framework.Logf("monitor[%s]: initialized", MonitorName)
+	return nil
+}
+
+func (test *monitorTest) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	return monitorapi.Intervals{}, nil, nil
+}
+
+func (*monitorTest) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	return nil, nil
+}
+
+func (test *monitorTest) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	intervals := test.monitor.Filter(finalIntervals)
+	if len(intervals) == 0 {
+		return nil, nil
+	}
+	framework.Logf("monitor[%s]: found %d interesting intervals ", MonitorName, len(intervals))
+
+	callback := &junitCallback{name: "[sig-apimachinery] conections (both reused and new) to the kube-apiserver during the graceful termination process should be handled gracefully"}
```

Comment:
s/conections/connections

Suggest: [sig-apimachinery] new and reused kube-apiserver connections should be handled gracefully during the graceful termination process